pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1217,6 +1222,31 @@ func (og *operationGenerator) GenerateUnmapVolumeFunc(
 		podDeviceUnmapPath, volName := blockVolumeUnmapper.GetPodDeviceMapPath()
 		// plugins/kubernetes.io/{PluginName}/volumeDevices/{volumePluginDependentPath}/{podUID}
 		globalUnmapPath := volumeToUnmount.DeviceMountPath
+
+		devicePath, err := filepath.EvalSymlinks(filepath.Join(globalUnmapPath, string(volumeToUnmount.PodUID)))
+		if err != nil {
+			return volumeToUnmount.GenerateError("UnmapVolume failed", err)
+		}
+
+		// Search under globalUnmapPath dir if all symbolic links except this pod have been removed already.
+		// If number of symbolic links are more than one, another pods may still refer the volume.
+		refs, err := og.blkUtil.GetDeviceSymlinkRefs(devicePath, globalUnmapPath)
+		if err != nil {
+			return volumeToUnmount.GenerateError("UnmapVolume.GetDeviceSymlinkRefs check failed", err)
+		}
+		if len(refs) > 1 {
+			err = fmt.Errorf("The device %q is still referenced from other Pods %v", globalUnmapPath, refs)
+			return volumeToUnmount.GenerateError("UnmapVolume failed", err)
```

Comment:
we could DetachFileDevice only if len(refs) == 1 but still proceed to UnmapPodDevice if len(refs) > 1?

Reply:
No, that won't work. Continuing discussion in original thread...